theano/compile/function_module.py

```diff
@@ -714,7 +714,13 @@ def checkSV(sv_ori, sv_rpl):
 
         f_cpy = maker.__class__(inputs=ins, outputs=outs, fgraph=fg_cpy,
                                 mode=maker.mode, profile=profile,
-                                on_unused_input=maker.on_unused_input,
+                                # When removing updates containing variables
+                                # not used in the output function, copy
+                                # generates an unused implicit input.
+                                # We ignore the resulting errors,
+                                # but could change it to 'warn' if this might
+                                # cause problems.
+                                on_unused_input='ignore',
```

Comment:
This is good and safe. But ideally, we should remove the shared variable from inputs. For now, we can keep it like this.